### PR TITLE
misc: Add grading rules from ocw

### PR DESCRIPTION
**Pull request by Alex-deVis**
Add initial version of the rules for the Operating Systems course inspired by the previous version on OCW.
This version contains TODOs and TBAs.

### CODE REVIEW

File: content/misc/rules-and-grading.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+1. Open-Book Interview:
+
+    - It will last for **15 minutes for each student**.
+    - Each student will know his time slot prior to the exam.
+    - Each student will be asked questions by one or both examinators and he will be awarded points based on his answers.
+    <!-- TODO: Is exam bonus still a thing? -->
+    - The maximum score is 11 (out of 10), with 1 point as a bonus.
+    <!-- TODO: Update percentage and points -->
+    - This component accounts for **70% of the total exam score**, which is **2.8 out of the total 4 points**.
+
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:36]:
```suggestion

```

File: content/misc/rules-and-grading.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    - Each student will know his time slot prior to the exam.
+    - Each student will be asked questions by one or both examinators and he will be awarded points based on his answers.
+    <!-- TODO: Is exam bonus still a thing? -->
+    - The maximum score is 11 (out of 10), with 1 point as a bonus.
+    <!-- TODO: Update percentage and points -->
+    - This component accounts for **70% of the total exam score**, which is **2.8 out of the total 4 points**.
+  
+    <!-- TODO: Find a more suggestive name -->
+1. Concepts Understanding
+  - It will last for **20 minutes**.
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:39]:
```suggestion

  - It will last for **20 minutes**.
```

File: content/misc/rules-and-grading.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    - It will last for **15 minutes for each student**.
+    - Each student will know his time slot prior to the exam.
+    - Each student will be asked questions by one or both examinators and he will be awarded points based on his answers.
+    <!-- TODO: Is exam bonus still a thing? -->
+    - The maximum score is 11 (out of 10), with 1 point as a bonus.
+    <!-- TODO: Update percentage and points -->
+    - This component accounts for **70% of the total exam score**, which is **2.8 out of the total 4 points**.
+  
+    <!-- TODO: Find a more suggestive name -->
+1. Concepts Understanding
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:38]:
```suggestion
1. Understanding Concepts
```

File: content/misc/rules-and-grading.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+  - It will last for **20 minutes**.
+  - This part is also **open-book**, meaning you can use written materials and offline resources.
+  However, internet access will not be allowed.
+  - All students scheduled for the exam in the same day will take this component simultaneously, and it will be conducted on [Moodle](https://curs.upb.ro/2023/course/view.php?id=7774).
+  - You will receive **11 True/False questions** that require explanations.
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:43]:
```suggestion
  - You will receive **11 True/False questions** that require explanations.
  - You will receive 0 points if you choose the right answer, but the explanation is wrong.
```

File: content/misc/rules-and-grading.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    <!-- TODO: Is exam bonus still a thing? -->
+    - The maximum score is 11 (out of 10), with 1 point as a bonus.
+    <!-- TODO: Update percentage and points -->
+    - This component accounts for **70% of the total exam score**, which is **2.8 out of the total 4 points**.
+  
+    <!-- TODO: Find a more suggestive name -->
+1. Concepts Understanding
+  - It will last for **20 minutes**.
+  - This part is also **open-book**, meaning you can use written materials and offline resources.
+  However, internet access will not be allowed.
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:41]: Isn't it? Asking questions isn't allowed, but using the internet in a read-only manner is.

File: content/misc/rules-and-grading.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+1. [Recourdings OS 2019-2020, 2020-2021 3CC](https://web.microsoftstream.com/channel/65910ece-e9a5-4a88-8641-cec8816ee99a?referrer=https:%2F%2Focw.cs.pub.ro%2F )
+1. [OS oral exam simulation 2022-2023](https://ctipub.sharepoint.com/sites/03acsla2s1socacbcccd/_layouts/15/stream.aspx?id=%2Fsites%2F03acsla2s1socacbcccd%2FShared%20Documents%2FGeneral%2FRecordings%2FSO%202022%2D2023%20%2D%20Simulare%20examen%20oral%2D20230121%5F165841%2DMeeting%20Recording%2Emp4&ga=1)
+1. `Concepts Understanding` exam model - **TBA**
+1. [Course notes](https://drive.google.com/drive/folders/1T08PmJ_fKTA8FLHvhXYw33CBK6D0QPpR)
+
+### [Assignments](https://cs-pub-ro.github.io/operating-systems/Assignments/)
+
+<!-- TODO: Update points -->
+Homework assignments are worth **TODO points** of the final grade.
+In total, there are 5 assignments, 2 of which are bonus.
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:77]:
```suggestion
There are 3 assignments, followed by 2 bonus ones.
```
Such that we never state the obvious: that there are 5 assignments.

File: content/misc/rules-and-grading.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+Each lab is graded between **0 and 12 points** and the final grade is calculated by summing up all scores and dividing to the number of labs.
+The score is rounded to the nearest whole number.
+
+The lab grade is composed of the following components:
+
+- 3 points for the **Moodle lab quiz**, which includes 4 questions: 2 questions from the preceding lab and 2 from the current one.
+- 2 points for involvement, awarded by the teaching assistant
+- 3 points for completing the **basic exercises** of the laboratory
+- 2 points for completing the **advanced exercises** of the laboratory
+- 2 points for completing the **bonus exercises** of the laboratory
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:95]:
```suggestion
- 2 points for **bonus activity** during the laboratory.
They are awarded by the TA according to their own criteria: interesting questions, good answers, solving bonus tasks, helping other students, etc.
```

File: content/misc/rules-and-grading.md [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+
+If you missed a lab or did not receive the maximum score on other labs, the lab bonuses will assist you in improving your final grade.
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:97]:
```suggestion
If you miss a lab or do not receive the maximum score on other labs, the lab bonuses will assist you in improving your final grade.
```

File: content/misc/rules-and-grading.md [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+**You can recover up to 2 missed labs with the agreement of the lab assistant.**
+
+**Collaboration during lab sessions is encouraged.**
```
Comment (teodutu) [inline content/misc/rules-and-grading.md:100]:
```suggestion
**Collaboration during lab sessions is strongly encouraged.**
```

**Review by teodutu — CHANGES_REQUESTED** [review]
I made a partial review

File: content/misc/rules-and-grading.md [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+    - Series CB: **TBA**
+    - Series CC: **TBA**
+    - Series CD: **TBA**
+
+1. Quiz 2 - covering questions from the [Compute chapter](https://cs-pub-ro.github.io/operating-systems/Lab/Compute/), will take place on the following dates:
+
+    - Series CA: **TBA**
+    - Series CB: **TBA**
+    - Series CC: **TBA**
+    - Series CD: **TBA**
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:119]:
```suggestion
    - Series CA: 07.11.2023
    - Series CB: 08.11.2023
    - Series CC: 07.11.2023
    - Series CD: 07.11.2023
```

File: content/misc/rules-and-grading.md [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+
+1. Quiz 3 - covering questions from the [I/O chapter](https://cs-pub-ro.github.io/operating-systems/Lab/IO/), will take place on the following dates:
+
+    - Series CA: **TBA**
+    - Series CB: **TBA**
+    - Series CC: **TBA**
+    - Series CD: **TBA**
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:126]:
```suggestion
    - Series CA: 05.12.2023
    - Series CB: 06.12.2023
    - Series CC: 05.12.2023
    - Series CD: 05.12.2023
```

File: content/misc/rules-and-grading.md [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+
+1. Quiz 4 - covering questions from the **TODO**, will take place on the following dates:
+
+    - Series CA: **TBA**
+    - Series CB: **TBA**
+    - Series CC: **TBA**
+    - Series CD: **TBA**
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:133]:
```suggestion
    - Series CA: 09.01.2024
    - Series CB: 10.01.2024
    - Series CC: 09.01.2024
    - Series CD: 09.01.2024
```

File: content/misc/rules-and-grading.md [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+
+The quizzes will be held on the Moodle platform during the class and will last for **5 minutes**.
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:135]:
```suggestion
The quizzes  are closed-book, will be held on the Moodle platform during the class and will last for **5 minutes**.
```

File: content/misc/rules-and-grading.md [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+During the semester, students will be rewarded with distinctions.
+
+The distinctions to be awarded in the 2023-2024 academic year are as follows:
+
+<!-- TODO: Complete with actual numbers -->
+- **TBA** distinctions for involvement in course activities.
+- **TBA** distinctions for involvement in laboratory activities.
+- **TBA** distinctions for interesting implementations of assignments.
+- **TBA** distinctions for exam results.
+- **TBA** distinctions for community involvement (pull requests, forum discussions, etc.).
```
Comment (elena19m) [inline content/misc/rules-and-grading.md:240]:
```suggestion
- 8 distinctions for involvement in course activities.
- 20 distinctions for involvement in laboratory activities.
- 5 distinctions for interesting implementations of assignments.
- 8 distinctions for exam results.
- 10 distinctions for community involvement (pull requests, forum discussions, etc.).
```

**Review by elena19m — COMMENTED** [review]
I've added suggestions for the TODOs in the course quizzes and pins sections.

File: content/misc/rules-and-grading.md [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+Starting from the third week, the attendance list for the laboratory becomes fixed.
+
+Transfers between subgroups are only allowed in objective situations.
+Reasons such as "I'm employed," "I have classes for another course," or "the schedule suits me better" are not considered.
+If you have an objective situation, you can request a transfer between subgroups by sending an email to [Răzvan Deaconescu](mailto:razvan.deaconescu@upb.ro) with the subject `[SO][Transfer Request] NAME Surname - Group`, in which you describe the situation.
+You will be informed if the transfer is accepted.
+
+### Pins
+
+During the semester, students will be rewarded with distinctions.
```
Comment (fabianpatras) [inline content/misc/rules-and-grading.md:231]:
```suggestion
After the semester, students will be rewarded with distinctions based on their involvement in certain activities.
```

**Comment by fabianpatras** [comment]
> Published at http://open-education-hub.github.io/operating-systems/2/

This gives 404.

**Comment by razvand** [comment]
@Alex-deVis, I'll check this later today and be back with feedback, so we can then merge it.

File: content/misc/rules-and-grading.md [hunk 16]
```diff
@@ -0,0 +1,271 @@
...
+Avoid testing an assignment on a classmate's system.
+There can be exceptions, such as assisting someone with troubleshooting, but please be mindful not to transition from "let's figure out the problem" to "let me solve your assignment."
+We recommend using the dedicated forum for each assignment or the MSTeams channel to ask questions (or any other channel provided to you).
+
+### Plagiarism Penalties
+
+Generally, we consider punitive measures to be the last resort.
+As long as an assignment is completed individually without problematic source code contributions from external sources, it is not considered a copied assignment.
+
+The concept of copied assignments includes, but is not limited to, situations such as:
```
Comment (razvand) [inline content/misc/rules-and-grading.md:198]:
```suggestion
The concept of plagiarized assignments includes, but is not limited to, situations such as:
```

File: content/misc/rules-and-grading.md [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+- Two assignments that bear sufficient resemblance to draw this conclusion.
+- Using source code from the internet that is clearly the solution to the assignment.
+- Utilizing portions of another classmate's code.
+- Accessing another classmate's code during the assignment.
+- Modifying an existing assignment.
+- Tracing another classmate's code.
+- Direct assistance in completing the assignment (someone else wrote or dictated the code).
+- Someone else completes the assignment (willingly, for payment, or other benefits).
+
+In the event that two assignments are deemed copied, both the source and destination will receive identical penalties, without discussions regarding who copied from whom and whose fault it is.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:209]:
```suggestion
In the event that two assignments are deemed plagiarized, both the source and destination will receive identical penalties, without discussions regarding who copied from whom and whose fault it is.
```

File: content/misc/rules-and-grading.md [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+
+Copying any part of an assignment results in the entire assignment's score being voided.
+**Any assignment submitted (before or after the plagiarized assignment) in the same academic session receives a score of 0 and is not eligible for resubmission within the current academic year.**
+
+We emphasize that our goal is not and will not be to penalize for copying.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:214]:
```suggestion
We emphasize that our goal is not and will not be to penalize for plagiarism.
```

File: content/misc/rules-and-grading.md [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+We view copying as a dishonest act that will be punished if it occurs.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:215]:
```suggestion
We view plagiarism as a dishonest act that will be punished if it occurs.
```

File: content/misc/rules-and-grading.md [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+Our objective, however, is to prevent copying, and to achieve this, we provide support and resources from our team in all their forms.
+
+### Laboratory
+
+The laboratory sessions are conducted in a physical format in the rooms assigned according to the schedule.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:220]:
```suggestion
Laboratory sessions take part in person in the lab rooms assigned according to the schedule.
```

File: content/misc/rules-and-grading.md [hunk 21]
```diff
@@ -0,0 +1,271 @@
...
+The groups are divided in half (into two equally-sized subgroups or with a maximum difference of one student).
```
Comment (razvand) [inline content/misc/rules-and-grading.md:221]:
```suggestion
Student groups are divided in half (into two equally-sized subgroups or with a maximum difference of one student).
```

File: content/misc/rules-and-grading.md [hunk 22]
```diff
@@ -0,0 +1,271 @@
...
+**The maximum number of students in a laboratory is 16.**
```
Comment (razvand) [inline content/misc/rules-and-grading.md:222]:
```suggestion
**The maximum number of students in a lab slot is 16.**
```

File: content/misc/rules-and-grading.md [hunk 23]
```diff
@@ -0,0 +1,271 @@
...
+
+- 8 distinctions for involvement in course activities.
+- 20 distinctions for involvement in laboratory activities.
+- 5 distinctions for interesting implementations of assignments.
+- 8 distinctions for exam results.
+- 10 distinctions for community involvement (pull requests, forum discussions, etc.).
+
+### Retaking the Final Exam
+
+In the resit/improvement session in the fall and in the special summer session (**only for 4th-year students**), only the exam can be retaken, which consists of the two components: the open-book interview and the written synthesis part.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:244]:
```suggestion
During the autumn exam and the special summer session (**only for 4th-year students**), only the exam can be retaken.
The exam consists of the two components: the open-book interview and the written synthesis part.
```

File: content/misc/rules-and-grading.md [hunk 24]
```diff
@@ -0,0 +1,271 @@
...
+The coursework assignments are not retaken.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:245]:
```suggestion
```

File: content/misc/rules-and-grading.md [hunk 25]
```diff
@@ -0,0 +1,271 @@
...
+Assignments cannot be submitted during the vacation period.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:246]:
```suggestion
Assignments cannot be resubmitted.
```

File: content/misc/rules-and-grading.md [hunk 26]
```diff
@@ -0,0 +1,271 @@
...
+
+You can participate in the exam to improve the grade obtained in the summer session.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:248]:
```suggestion
You can reatek exam to improve the grade obtained in the first exam session.
```

File: content/misc/rules-and-grading.md [hunk 27]
```diff
@@ -0,0 +1,271 @@
...
+In the case of participation for grade improvement, the grade obtained in this manner will replace the previously obtained grade from the summer session.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:249]:
```suggestion
In the case of participation for grade improvement, the grade obtained in this manner will replace the grade in the first exam session.
```

File: content/misc/rules-and-grading.md [hunk 28]
```diff
@@ -0,0 +1,271 @@
...
+
+Students in the 4th year can also take the exams in the resit/improvement session in the fall and the special summer session.
+4th-year students can participate in both the summer and fall sessions.
+
+You can participate in only one exam during a session.
+Once the exam dates are established (generally, there are two exam dates in the resit/improvement session), we will open discussion threads for registration for one of the two dates.
+Then we will plan and announce the details.
+Grades will be recorded in the official register, following the course's register.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:257]:
```suggestion
```

File: content/misc/rules-and-grading.md [hunk 29]
```diff
@@ -0,0 +1,271 @@
...
+If, after the resit/improvement session in the fall, you still do not obtain a passing score, you will need to retake the course in the next academic year.
+
+### Retaking the Course
+
+By default, the grades for the 2023-2024 academic year is reset on October 1, 2024.
+If you do not graduate from the course in the 2023-2024 academic year, you will need to retake it in the 2024-2025 academic year, with the possibility of retaining some components, as detailed below.
+
+- If you wish, you can retain the grades for the course's lab components (laboratory and/or assignments).
```
Comment (razvand) [inline content/misc/rules-and-grading.md:265]:
```suggestion
- If you wish, you can retain the grades for the course's practical components (laboratory and/or assignments).
```

File: content/misc/rules-and-grading.md [hunk 30]
```diff
@@ -0,0 +1,271 @@
...
+You **cannot** retain the grades for the course components (coursework and the final exam).
```
Comment (razvand) [inline content/misc/rules-and-grading.md:266]:
```suggestion
You **cannot** retain the grades for lecture components (lecture tests and exam).
```

File: content/misc/rules-and-grading.md [hunk 31]
```diff
@@ -0,0 +1,271 @@
...
+- Students interested in preserving their grades should respond on a designated Moodle thread by **TBA**.
+- The rules and grading system for students retaking the course are the same as for students in the current year (course quizzes + exam, homework assignments, etc.).
```
Comment (razvand) [inline content/misc/rules-and-grading.md:268]:
```suggestion
- The rules and grading system for students retaking the course are the same as for students in the current year (lecture tests + exam, assignments, etc.).
```

File: content/misc/rules-and-grading.md [hunk 32]
```diff
@@ -0,0 +1,271 @@
...
+- Students retaking the course can participate in any lab session as long as there are available slots.
+Like other students, the attendance list becomes fixed starting from the third week.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:270]:
```suggestion
  Like other students, the attendance list becomes fixed starting from the third week.
```

File: content/misc/rules-and-grading.md [hunk 33]
```diff
@@ -0,0 +1,271 @@
...
+- Students retaking the course can participate in any lecture, course test, and final exam date, regardless of the series to which they belonged.
```
Comment (razvand) [inline content/misc/rules-and-grading.md:271]:
```suggestion
- Students retaking the course can participate in any lecture, lecture test, and exam, regardless of the series to which they belonged.
```

**Review by razvand — CHANGES_REQUESTED** [review]
See my comments.

File: content/misc/rules-and-grading.md [hunk 34]
```diff
@@ -0,0 +1,271 @@
...
+
+<!-- TODO: Update total points -->
+To promote this course you need to obtain  **4.5 points** from a total of **11 points**:
+
+<!-- TODO: Update components' points -->
+- 4 points for the [final exam](#final-exam)
+- 5 points for [assignments](#assignments)
+- 1 point for [laboratory sessions](#laboratory-sessions)
+- 1 point for [course quizzes](#course-quizzes)
+
```
[the remark this answers is not in the document]
Comment (Alex-deVis) [reply content/misc/rules-and-grading.md:11]: This is already stated below, under #exam:

> **Passing the course requires obtaining a minimum of 2 out of 4 points (i.e., 50%) of the exam score.**

File: content/misc/rules-and-grading.md [hunk 35]
```diff
@@ -0,0 +1,271 @@
...
+The Operating Systems exam for this academic year, 2023-2024, is worth **4 points** and consists of two components:
+
+1. Open-Book Interview:
+
+    - It will last for **15 minutes for each student**.
+    - Each student will know his time slot prior to the exam.
+    - Each student will be asked questions by one or both examinators and he will be awarded points based on his answers.
+    <!-- TODO: Is exam bonus still a thing? -->
+    - The maximum score is 11 (out of 10), with 1 point as a bonus.
```
[the remark this answers is not in the document]
Comment (Alex-deVis) [reply content/misc/rules-and-grading.md:29]: I checked and the recommended [markdown indentation for sublists is **4 spaces**](https://docs.newrelic.com/docs/style-guide/structure/styleguide-markup-indentation/#four-space).

**Comment by Alex-deVis** [comment]
@razvand I applied most of your comments except for the ones I replied to. 
I am waiting for your feedback on the last two TODO items:
- What is the minimum grade to skip the exam (if this idea materializes)
- What is the score for each assignment?

File: content/misc/rules-and-grading.md (hunk 5 above)
Comment (Alex-deVis) [reply content/misc/rules-and-grading.md]: @teodutu looks like the internet is a thing of the past.

**Review by razvand — APPROVED** [review]
All good.

**Pull request merged**